iiif_prezi3/helpers/set_hwd.py

```diff
@@ -0,0 +1,29 @@
+from ..loader import monkeypatch_schema
+from ..skeleton import Canvas, ResourceItem
+
+
+class SetHwd:
+    def set_hwd(self, height, width, duration):
+        """Set the height, width, and duration properties allowing nulls.
+
+        Args:
+            height (int): The height of the resource or the canvas
+            width (int): The width of the resource or the canvas
+            duration (float): The duration of the resource
+
+        Returns:
+            None
+        """
+        if duration is None and height is None and width is None:
+            raise TypeError("At least one of height, width, or duration must be set")
+        if height is not None and width is None:
+            raise TypeError("width must be set if height is set")
+        if width is not None and height is None:
+            raise TypeError("height must be set if width is set")
```

Comment:
Is zero allowed, or any other empty values? Or is important to check explicitly for `None`?

If not, could just do a boolean check here, e.g. `if not (height or width or duration)`

Reply:
The spec doesn't explicitly say width and height must be non-zero, but then it also doesn't say they *must* be positive integers either.

@azaroth42 @glenrobson ?

Reply:
I suppose there could be an instance where a user tries to set H+W to 0 for an Audio resource, which would _technically_ be correct, I suppose?

Reply:
The spec says "The value must be a positive integer." so I think it implies its 1 or over...

Reply:
@glenrobson  @rlskoeser  @digitaldogsbody In the json schema however the minimum for duration is 0: 
https://github.com/IIIF/presentation-validator/blob/bd7468500f8ebfa82aa54f6c80ddd4f22f20c71a/schema/iiif_3_0.json#L62

Reply:
> The spec says "The value must be a positive integer." so I think it implies its 1 or over...

Somehow my eyes completely bypassed that sentence when I was reading the spec :joy: 

Is there a way to enforce this in the Schema? That would seem the best place to do so, but if it's not possible, then we should add it here in the helper.

Reply:
@digitaldogsbody  i think that `"minimum": 1`  should do the trick!
https://github.com/IIIF/presentation-validator/blob/bd7468500f8ebfa82aa54f6c80ddd4f22f20c71a/schema/iiif_3_0.json#L62

Reply:
We'd also need to add a minimum for the height and width:
https://github.com/IIIF/presentation-validator/blob/bd7468500f8ebfa82aa54f6c80ddd4f22f20c71a/schema/iiif_3_0.json#L141
https://github.com/IIIF/presentation-validator/blob/bd7468500f8ebfa82aa54f6c80ddd4f22f20c71a/schema/iiif_3_0.json#L531

One thing we should check is if a minimum value of 1 will prevent blank values (because of an evaluation to 0). I don't know if this is the case with how Pydantic works (I suspect not) but it might be for some JSON Validators?

The potential issue is that for an audio file, H & W = 0 would _technically_ be correct, although it would violate the wording of the spec (if we read "positive integer" as >= 1, that is), so we should maybe have a chat about how we want to handle this.

Reply:
@azaroth42 @zimeon @jpstroop - any thoughts on this as editors?